examples/rl_gsm8k/cot_math_agent.py

```diff
@@ -62,21 +67,35 @@ def parse_completion(self, completion: str, prompt_id: str) -> Generator[Step, N
             return
         yield step
 
+    def make_prompt(self, agent: Any, tape: Tape) -> Prompt:
+        messages = []
+        if self.system_prompt:
+            messages.append({"role": "system", "content": self.system_prompt})
+        messages.append({"role": "user", "content": tape.steps[0].llm_view()})
+        #messages = self.tape_to_messages(cleaned_tape, steps_description)
+        prompt_token_ids = agent.llm.tokenizer.apply_chat_template(
+            messages, add_special_tokens=True, add_generation_prompt=True
+        )
+        prompt_token_ids = prompt_token_ids[-self.max_prompt_length :]
+        return Prompt(messages=messages, token_ids=prompt_token_ids)
+
 
 #### Agent and Environment ####
 class CoTMathAgent(Agent):
     @classmethod
-    def create(cls, llm: LLM):
+    def create(cls, system_prompt: str, llm: LLM, max_prompt_length: int):
         agent = super().create(
             llm,
             nodes=[
                 ReasoningNode(
                     name="cot",
                     agent_step_cls=MathAgentStep,
-                    store_llm_calls=True,
+                    system_prompt=system_prompt if system_prompt else "",
```

Comment:
`system_prompt` is mandatory argument in this function, what is the purpose of this check?